### ISSUE TITLE
Error while creating a container from tensorflow-1.14.0 image on a ppc64le architecture machine

### ISSUE THREAD

**Issue by vnjoopud** [containers/libpod#3955]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

We have installed podman (version : 1.5.1) on a **ppc64le** architecture machine. We were able to create a container from a simple ubuntu image (ppc64le/ubuntu). But, we ran into an error while creating a container from tensorflow-1.14.0 image (ibmcom/tensorflow-ppc64le:1.14.0-gpu). 

**Steps to reproduce the issue:**

1. podman pull ibmcom/tensorflow-ppc64le:1.14.0-gpu

2. podman run --rm -it docker.io/ibmcom/tensorflow-ppc64le:1.14.0-gpu

**Describe the results you received:**
We get the following error message while executing the command `podman run --rm -it docker.io/ibmcom/tensorflow-ppc64le:1.14.0-gpu`

Error: time="2019-09-05T16:20:12-04:00" level=warning msg="signal: killed"
time="2019-09-05T16:20:12-04:00" level=error msg="container_linux.go:346: starting container process caused \"process_linux.go:449: container init caused \\\"process_linux.go:432: running prestart hook 0 caused \\\\\\\"error running hook: exit status 1, stdout: , stderr: exec command: [/usr/bin/nvidia-container-cli --load-kmods configure --ldconfig=@/sbin/ldconfig.real --no-cgroups --device=all --compute --utility --require=cuda>=10.0 --pid=46774 /tmp/vnjoopud/containers/storage/vfs/dir/5d77216afaecf0d51ae3b7650844a3369979034faa9e2c9927028aecb9cccef8]\\\\\\\\nnvidia-container-cli: initialization error: cuda error: unknown error\\\\\\\\n\\\\\\\"\\\"\"\n"

container_linux.go:346: starting container process caused "process_linux.go:449: container init caused \"process_linux.go:432: running prestart hook 0 caused \\\"error running hook: exit status 1, stdout: , stderr: exec command: [/usr/bin/nvidia-container-cli --load-kmods configure --ldconfig=@/sbin/ldconfig.real --no-cgroups --device=all --compute --utility --require=cuda>=10.0 --pid=46774 /tmp/_userName_/containers/storage/vfs/dir/5d77216afaecf0d51ae3b7650844a3369979034faa9e2c9927028aecb9cccef8]\\\\nnvidia-container-cli: initialization error: cuda error: unknown error\\\\n\\\"\"": OCI runtime error

**Describe the results you expected:**
We expected to create a container (without any errors) from the above image.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:            1.5.1
RemoteAPI Version:  1
Go Version:         go1.10.4
OS/Arch:            linux/ppc64le

```

**Output of `podman info --debug`:**

```
debug:
  compiler: gc
  git commit: ""
  go version: go1.10.4
  podman version: 1.5.1
host:
  BuildahVersion: 1.10.1
  Conmon:
    package: 'conmon: /usr/libexec/podman/conmon'
    path: /usr/libexec/podman/conmon
    version: 'conmon version 2.0.0, commit: unknown'
  Distribution:
    distribution: ubuntu
    version: "18.04"
  MemFree: 357939478528
  MemTotal: 548435066880
  OCIRuntime:
    package: 'cri-o-runc: /usr/lib/cri-o-runc/sbin/runc'
    path: /usr/lib/cri-o-runc/sbin/runc
    version: 'runc version spec: 1.0.1-dev'
  SwapFree: 23152492544
  SwapTotal: 23152492544
  arch: ppc64le
  cpus: 192
  eventlogger: journald
  hostname: _myMachineName_
  kernel: 4.15.0-58-generic
  os: linux
  rootless: true
  uptime: 47h 36m 30.75s (Approximately 1.96 days)
registries:
  blocked: null
  insecure: null
  search:
  - docker.io
store:
  ConfigFile: /home/_userName_/.config/containers/storage.conf
  ContainerStore:
    number: 0
  GraphDriverName: vfs
  GraphOptions: null
  GraphRoot: /tmp/_userName_/containers/storage
  GraphStatus: {}
  ImageStore:
    number: 3
  RunRoot: /run/user/2314
  VolumePath: /tmp/_userName_/containers/storage/volumes
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
Listing... Done
podman/bionic,now 1.5.1-1~ubuntu18.04~ppa1 ppc64el [installed]
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by mheon** [comment]
This appears to be an error in the Nvidia OCI hook, and not Podman itself

**Comment by rhatdan** [comment]
Yes you should report this issue to Nvidia and see if they have ideas.

**Comment by vnjoopud** [comment]
The nvidia OCI hooks were working with my previous version of podman on the same machine. Previously, we had podman (version 1.4.4) on the same machine and I was able to launch a container from this image (without any errors). We updated podman to 1.5.1 and then we noticed the above exception.

**Comment by rhatdan** [comment]
Any chance you can downgrade and see if the older version still works?

I do notice that you are running with vfs back end?  Any reason you are not using overlay? fuse_overlay?

**Comment by vnjoopud** [comment]
We were not able to downgrade to a lower version (from the current 1.5.1 version to 1.4.4) as we couldn't find the lower version of podman in the repo.

`
bash$: cat /etc/apt/sources.list.d/projectatomic-ubuntu-ppa-bionic.list
deb http://ppa.launchpad.net/projectatomic/ppa/ubuntu bionic main
`
In the above repo, I could only find 1.5.1 version of podman. 

Can you let me know about the instructions for downgrading to a lower version.

Also, one more thing, I would like to add is that we also upgraded our ubuntu (from 18.04.2 LTS to 18.04.3 LTS) and then we ran the update on podman (which went from 1.4.4 to 1.5.1)

(Thank you for the quick response)

**Comment by vnjoopud** [comment]
We configured storage in /tmp (did not specify vfs)

**Comment by rhatdan** [comment]
Does podman work if you run it as root?
If you install fuse-overlayfs and then cleanout your storage and storage config.  Podman will default to using fuse-overlay.

**Comment by vnjoopud** [comment]
I have a small question regarding it as to whether fuse-overlayfs will work if storage is on a NFS mounted directory.

```
root@:~# podman run --rm docker.io/ibmcom/tensorflow-ppc64le:1.14.0-gpu nvidia-smi
Failed to initialize NVML: Unknown Error
```

**Comment by rhatdan** [comment]
User Namespace will not work properly, but not sure why fuse-overlay would have an issue.
@giuseppe WDYT?

**Comment by giuseppe** [comment]
> I have a small question regarding it as to whether fuse-overlayfs will work if storage is on a NFS mounted directory.

fuse-overlayfs won't work on NFS.

Also I've not seen the "Failed to initialize NVML: Unknown Error" before, so I guess it is coming from the OCI hook?

**Comment by rhatdan** [comment]
@giuseppe yes @vnjoopud says that this was working on previous podman 1.4.4 version but is now blowing up on podman 1.5.1.

**Comment by baude** [comment]
Any chance this is it? -> https://github.com/Kaixhin/dockerfiles/issues/1

**Comment by vnjoopud** [comment]
@rhatdan : I wouldn't mind downgrading my podman version to 1.4.4. Can you let me know about the instructions for downgrading to a lower version. (Currently, I am not able to find the 1.4.4 version of podman in the repo. I posted more details regarding to this in my earlier comments.)

@baude : Even with the 1.4.4 version of podman, we are getting the same "NVML" error while running podman as root (but, the 1.4.4 version was able to create a container from tensorflow-1.4.0 image without any error.)
